staging/src/k8s.io/kube-apiextensions-server/pkg/apis/apiextensions/validation/validation.go

```diff
@@ -109,13 +109,14 @@ func ValidateCustomResourceDefinitionSpecUpdate(spec, oldSpec *apiextensions.Cus
 	allErrs := ValidateCustomResourceDefinitionSpec(spec, fldPath)
 
 	if established {
-		// these all affect the storage, so you can't change them
+		// this affects the resource name, which is always immutable, so this can't be updated.
 		genericvalidation.ValidateImmutableField(spec.Group, oldSpec.Group, fldPath.Child("group"))
```

Comment:
This needs to move out of the if block, because even if you haven't been established, you can't change this because of the affect on the resource name.